libs/MeshKernel/include/MeshKernel/Exceptions.hpp

```diff
@@ -87,15 +88,22 @@ namespace meshkernel
     class MeshKernelError : public std::exception
     {
     public:
-        /// @brief Class default constructor. It is deleted to prevent constructing errors without messages.
-        MeshKernelError() = delete;
+        /// @brief Class constructor parametrized optionally by the source location.
+        /// @param[in] source_location The source location.
+        MeshKernelError(std::source_location const& source_location = std::source_location::current())
+            : m_what(),
+              m_fmt_message(),
+              m_source_location(source_location)
+        {
+        }
 
         /// @brief Class constructor parametrized by a variadic error message and optionally the source location.
         /// @param[in] message         The variadic error message.
         /// @param[in] source_location The source location.
         MeshKernelError(VariadicErrorMessage const& message,
```

Comment:
Can any of you figure out why:
```
throw MeshKernelError({"some useful message id ={}", 5}); // (1) ambiguous
throw MeshKernelError(VariadicErrorMessage{"some useful message id ={}", 5)}; // (2) ok
throw MeshKernelError({"some useful message id ={} and value = {}", 5, 2.5}); // (3) ok
```
(1) compiler cannot pick a ctor when there is one variadic arg (see the other ctors in class)  
(2)  removes ambiguity  
(3) similar to (1), but has more than one variadic arg. Treatment done in (2) not required.

Reply:
For me, I do no know what the problem is. Even after being in the language for >10 years, I have not used variadic templates, and std::format has only recently been added to gcc